agasc/agasc.py

```diff
@@ -1,42 +1,178 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 import os
 from pathlib import Path
+import warnings
+from contextlib import ContextDecorator
 
 import numpy as np
 import numexpr
 import tables
-import warnings
 
 from Chandra.Time import DateTime
 from astropy.table import Table, Column
-from ska_helpers.utils import LazyDict
+from ska_helpers.utils import lru_cache_timed
+
+__all__ = ['sphere_dist', 'get_agasc_cone', 'get_star', 'get_stars',
+           'MAG_CATID_SUPPLEMENT', 'BAD_CLASS_SUPPLEMENT', 'get_supplement_table',
+           'disable_supplement']
+
+DISABLE_SUPPLEMENT_ENV = 'AGASC_DISABLE_SUPPLEMENT'
+MAG_CATID_SUPPLEMENT = 100
+BAD_CLASS_SUPPLEMENT = 100
+
+RA_DECS_CACHE = {}
+COMMON_DOC = """\
+If ``use_supplement`` is ``True`` and there is no ``AGASC_DISABLE_SUPPLEMENT``
+    environment variable set, then stars with available mag estimates or bad
+    star entries in the AGASC supplement are updated in-place in the output
+    ``stars`` Table:
+
+    - ``MAG_ACA`` and ``MAG_ACA_ERR`` are set according to the supplement.
+    - ``MAG_CATID`` (mag catalog ID) is set to ``MAG_CATID_SUPPLEMENT`` (100).
+    - If ``COLOR1`` is 0.7 or 1.5 then it is changed to 0.69 or 1.49 respectively.
+    - If ``AGASC_ID`` is in the supplement bad stars table then CLASS is set to
+      ``100 + <source>``, where ``<source>`` is the bad star source identifier.
+
+    The default ``agasc_file`` is ``<AGASC_DIR>/miniagasc.h5``, where
```

Comment:
Seems like the common doc section should also mention the filename of the agasc supplement.  I agree that we can probably get by without adding an option to control / specify that by name.

Reply:
Will do.